### ISSUE TITLE
smtp sending not possible from docker container

### ISSUE THREAD

**Issue by lonesomewalker** [grafana/grafana#24769]
**What happened**:
After setting up smtp credentials, even double checked, Grafana is not able to send emails.

**What you expected to happen**:
Entering config variables to env, setting up notifications by Grafana, test, working

**How to reproduce it (as minimally and precisely as possible)**:
On smtp server side:
```
postfix/submission/smtpd[28744]: connect from grafana.server[xxx.xxx.xxx.xxx]
postfix/submission/smtpd[28744]: Anonymous TLS connection established from grafana.server[xxx.xxx.xxx.xxx]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
postfix/submission/smtpd[28744]: disconnect from grafana.server[xxx.xxx.xxx.xxx] ehlo=2 starttls=1 auth=1 quit=1 commands=5
```

From Grafana docker:
```
grafana          | t=2020-05-17T00:09:44+0000 lvl=eror msg="Failed to send alert notifications" logger=context userId=1 orgId=1 uname=xxx error="Failed to send notification to email addresses: xx@xxxxx.xx: gomail: could not send email 1: gomail: invalid address \"xx@xxxxx.xx <xx@xxxxx.xx>\": mail: expected single address, got \"<xx@xxxxx.xx>\"" remote_addr=xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx
```

**Environment**:
- Grafana version: 6.7.1
- OS Grafana is installed on: docker

**Docker settings**:
    environment:
      GF_SMTP_ENABLED: "true"
      GF_SMTP_HOST: "smtp.server.tld:587"
      GF_SMTP_USER: "xx@xxxxx.xx"
      GF_SMTP_PASSWORD: "xxxxxxxxxx"
      GF_SMTP_SKIP_VERIFY: "true"
      GF_SMTP_FROM_ADDRESS: "xx@xxxxx.xx"
      GF_SMTP_FROM_NAME: "xx@xxxxx.xx"

**Comment by bergquist** [comment]
The email alert notifications are tested quite frequently. I suspect there is something in the configuration that causes this.

**Comment by lonesomewalker** [comment]
Yes, they might be tested against Google or other bigger SMTP solutions.
Unfortunately, this is no option for EU and GDPR...

**Comment by marefr** [comment]
@lonesomewalker are you running in kubernetes?

**Comment by lonesomewalker** [comment]
@marefr no, dedicated server with Ubuntu.

**Comment by LuigiClemente-Awin** [comment]
Having the same issue with Grafana 7, official docker image. Telnet connection works from inside the container.

Error message:

`t=2020-06-19T12:48:27+0000 lvl=eror msg="Failed to send alert notifications" logger=context userId=1 orgId=1 uname=admin error="Failed to send notification to email addresses: luigi.clemente@xxx 454 4.7.0 TLS not available due to local problem" remote_addr=10.xx.xx.xx`

Configuration

[smtp]
enabled = true
host = smtp-relay.xxxxxx:25
user =
password =
cert_file =
key_file =
skip_verify = false
from_address = systemsengineering@xxxxx
from_name = grafana-dev
ehlo_identity = grafana-dev.xxxxxxx

**Comment by LuigiClemente-Awin** [comment]
Just adding to my previous message.
I also tested running a fresh docker image (alpine) on a Debian docker host on prem (no AWS).
The only thing I changed was the smpt configuration, and I still get an error:

`t=2020-06-22T10:26:03+0000 lvl=eror msg="failed to send notification" logger=alerting.notifier uid= error="Failed to send notification to email addresses: luigi.clemente@awin.com: 454 4.7.0 TLS not available due to local problem"`

SMTP
`[smtp]                                                                                                                                                                       
enabled = true                                                                                                                     
host = xxxxx.xxxxx.xx:25                                                            
;user =                                                                          
;password =                                                                                                
;cert_file =                                                                                          
;key_file =                                                                                                                        
skip_verify = true                          
;from_address = admin@grafana.localhost                                         
;from_name = Grafana                                                                                        
;ehlo_identity = dashboard.example.com  `

**Comment by LuigiClemente-Awin** [comment]
**FYI**
SysAdmins in my company have deployed a new relay server with TLS enabled and the issue disappeared. 
It looks like the new Grafana + Alpine Docker container cannot work with outdated relays without TLS.
I suggest you add config options to disable TLS completely.

**Comment by marefr** [comment]
@LuigiClemente-Awin doesn't `skip_verify = true` resolve this already?

**Comment by LuigiClemente-Awin** [comment]
@marefr no, it makes no difference at all: it seems that in case the SMTP relay is very old and doesn't support TLS at all, the process fails immediately. This might not be trapped by your automated tests because you are not testing with an old enough relay server (or with a very specific configuration).

**Comment by marefr** [comment]
Duplicate of #24563

**Comment by marefr** [comment]
#24563 was fixed by #24574 and is included in Grafana v7.1+
